airflow/providers/google/cloud/example_dags/example_dataflow_sql.py

```diff
@@ -0,0 +1,60 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Example Airflow DAG for Google Cloud Dataflow service
+"""
+import os
+
+from airflow import models
+from airflow.providers.google.cloud.operators.dataflow import DataflowStartSqlJobOperator
+from airflow.utils.dates import days_ago
+
+GCP_PROJECT_ID = os.environ.get("GCP_PROJECT_ID", "example-project")
+
+BQ_SQL_DATASET = os.environ.get('DATAFLOW_BQ_SQL_DATASET', 'airflow_dataflow_samples')
+DATAFLOW_SQL_JOB_NAME = os.environ.get('DATAFLOW_SQL_JOB_NAME', "dataflow-sql")
+DATAFLOW_SQL_LOCATION = os.environ.get('DATAFLOW_SQL_LOCATION', 'us-west1')
+
+with models.DAG(
+    dag_id="example_gcp_dataflow_sql",
+    default_args={
+        "start_date": days_ago(1),
+    },
+    schedule_interval=None,  # Override to match your needs
+) as dag_sql:
+    start_sql = DataflowStartSqlJobOperator(
+        task_id="start_sql_query",
+        job_name=DATAFLOW_SQL_JOB_NAME,
+        query=f"""
+            SELECT
+                sales_region as sales_region,
+                count(state_id) as count_state
+            FROM
+                bigquery.table.`{GCP_PROJECT_ID}`.beam_samples.beam_table
```

Comment:
Who is going to create this dataset? Can we use a public dataset so that examples works for anyone?

Reply:
This bucket is created in system tests. https://github.com/apache/airflow/pull/8553/files#
Unfortunately, Dataflow SQL is not compatible with the public datasets I know.

I got the following error when I referred to the public dataset.

> Caused by: java.lang.UnsupportedOperationException: Field type 'NUMERIC' is not supported (field 'value')

Its error message from BigQuery console
![Screenshot 2020-04-27 at 22 52 46](https://user-images.githubusercontent.com/12058428/80419684-de796580-88d9-11ea-85bd-1d5473c13901.png)

Reply:
We strive to maintain a high level of reliability for GCP integration. All integrations that are developed by my team must have system tests. System tests are required criteria for internal review.

Reply:
@ibzib - is there a public dataset that could be used?

Reply:
> Unfortunately, Dataflow SQL is not compatible with the public datasets I know.

Yeah, the table will have to have a schema that is compatible with DF SQL. A canonical public BQ dataset seems like something we should definitely have in the docs, but I couldn't find one.

Reply:
Yeah Dataflow SQL doesn't support GEOGRAPHY or NUMERIC, but I'm sure there are many public datasets that don't use those types. `chicago_taxi_trips.taxi_trips looks like it will work.